WalletWasabi/BranchNBound/BranchAndBound.cs

```diff
@@ -0,0 +1,108 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.BranchNBound
+{
+	public class BranchAndBound : Selector
+	{
+		private Money _costOfHeader = Money.Satoshis(0);
+		private Money _costPerOutput = Money.Satoshis(0);
+		private int _bnbTryLimit = 5;
+		private Random _random = new();
+
+		private Money[]? UtxoSorted { get; set; }
+
+		public bool TryGetExactMatch(Money target, List<Money> availableCoins, out List<Money> selectedCoins)
+		{
+			selectedCoins = new List<Money>();
+			UtxoSorted = availableCoins.OrderByDescending(x => x.Satoshi).ToArray();
+			try
+			{
+				for (int i = 0; i < _bnbTryLimit; i++)
```

Comment:
What purpose does this serve?

Reply:
It came from a misunderstanding of the original whitepaper, here the tolerance should be put later on.
This try limit should be inside the recursive logic, trying to prevent reaching the memory limit.
But IMO it's not that easy to find a good limit, as the variaty of users with different specs, good or bad, is wide.
Some would hit the limit every time, some will never. Any idea to prevent an OutOfMemory- or StackOverflowException?

Reply:
> It came from a misunderstanding of the original whitepaper,

Could you link the paper here?

> Any idea to prevent an OutOfMemory- or StackOverflowException?

Yes, remove recursion. Then it should be much harder to get to `StackOverflowException`.

Reply:
https://murch.one/wp-content/uploads/2016/11/erhardt2016coinselection.pdf
From page 37-40